dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/configurator/AbstractConfigurator.java

```diff
@@ -126,7 +126,25 @@ private URL configureIfMatch(String host, URL url) {
                 String currentApplication = url.getParameter(APPLICATION_KEY, url.getUsername());
                 if (configApplication == null || ANY_VALUE.equals(configApplication)
                         || configApplication.equals(currentApplication)) {
-                    Set<String> conditionKeys = new HashSet<String>();
+
+                    String tildeKey = null;
+                    for (Map.Entry<String, String> entry : configuratorUrl.getParameters().entrySet()) {
+                        String key = entry.getKey();
+                        String value = entry.getValue();
+
+                        if (StringUtils.isNotEmpty(key) && key.startsWith(TILDE)) {
+                            tildeKey = key;

```

Comment:
is it possible that there are more than one key match `TILDE`

Reply:
yes, i have modified it